PostHog/PostHogSDK.swift

```diff
@@ -831,6 +847,146 @@ let maxRetryDelay = 30.0
+    /// and will not include any locally overridden properties.
+    ///
+    /// ## Example Usage
+    /// ```swift
+    /// // Clear all locally set person properties for flags
+    /// PostHogSDK.shared.resetPersonPropertiesForFlags()
+    ///
+    /// // Feature flags will now use only server-side properties
+    /// let flagValue = PostHogSDK.shared.isFeatureEnabled("feature")
+    /// ```
+    ///
+    /// - Note: This method does not automatically reload feature flags. Call `reloadFeatureFlags()`
+    ///         after resetting if you want to immediately refresh flags with the cleared properties.
+    @objc public func resetPersonPropertiesForFlags() {
+        if !isEnabled() {
+            return
+        }
+
+        remoteConfig?.resetPersonPropertiesForFlags()
+    }
+
+    /// Sets properties for a specific group type to include when evaluating feature flags.
+    /// These properties supplement the standard group information sent to PostHog for flag evaluation,
+    /// providing additional context that can be used in flag targeting conditions.
+    ///
+    /// ## Example Usage
+    /// ```swift
+    /// PostHogSDK.shared.setGroupPropertiesForFlags("organization", properties: [
+    ///     "plan": "enterprise",
+    ///     "seats": 50,
+    ///     "industry": "technology"
+    /// ])
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - groupType: The group type identifier (e.g., "organization", "team")
+    ///   - properties: Dictionary of properties to set for this group type
+    /// - Note: This method automatically reloads feature flags to apply the new properties.
+    @objc public func setGroupPropertiesForFlags(_ groupType: String, properties: [String: Any]) {
+        if !isEnabled() {
+            return
+        }
+
+        let sanitizedProperties = sanitizeDictionary(properties) ?? [:]
+        remoteConfig?.setGroupPropertiesForFlags(groupType, properties: sanitizedProperties)
+
+        // Automatically reload flags to apply the new properties
+        remoteConfig?.reloadFeatureFlags()
+    }
+
+    /// Clears group properties for feature flag evaluation for a specific group type,
+    /// or all group properties if no group type is specified.
+    ///
+    /// ## Example Usage
+    /// ```swift
+    /// // Clear properties for specific group type
+    /// PostHogSDK.shared.resetGroupPropertiesForFlags("organization")
+    ///
+    /// // Clear all group properties
+    /// PostHogSDK.shared.resetGroupPropertiesForFlags()
+    /// ```
+    ///
+    /// - Parameter groupType: The group type to clear properties for, or nil to clear all group properties
+    /// - Note: This method does not automatically reload feature flags. Call `reloadFeatureFlags()`
+    ///         after resetting if you want to immediately refresh flags with the cleared properties.
+    @objc public func resetGroupPropertiesForFlags(_ groupType: String? = nil) {
```

Comment:
@ioannisj do we need a method overload here for better objC uX? otherwise they have to forcefully pass (nil)